### ISSUE TITLE
Swarm can list some nodes but I can't see them on info

### ISSUE THREAD

**Issue by pablokbs** [docker/swarm#1647]
Hello, I'm trying to add two more nodes to my already working swarm but I can't.

I see my original two nodes here:

```
root@jenkins-master:~# docker -H tcp://127.0.0.1:3241 info
Containers: 7896
Images: 29
Storage Driver:
Role: primary
Strategy: spread
Filters: health, port, dependency, affinity, constraint
Nodes: 2
 jenkins-slave-1: 10.0.1.143:4243
  └ Status: Healthy
  └ Containers: 3948
  └ Reserved CPUs: 0 / 8
  └ Reserved Memory: 0 B / 15.42 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-48-generic, operatingsystem=Ubuntu 14.04.2 LTS, storagedriver=devicemapper
 jenkins-slave-2: 10.0.1.86:4243
  └ Status: Healthy
  └ Containers: 3948
  └ Reserved CPUs: 0 / 8
  └ Reserved Memory: 0 B / 15.42 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-48-generic, operatingsystem=Ubuntu 14.04.2 LTS, storagedriver=devicemapper
Execution Driver:
Kernel Version:
Operating System:
CPUs: 16
Total Memory: 30.84 GiB
Name: a06db689250e
ID:
Http Proxy:
Https Proxy:
No Proxy:
```

I see the four nodes running swarm list:

```
root@jenkins-master:~# docker run --rm swarm list token://xxxx
10.0.1.118:4243
10.0.1.143:4243
10.0.1.119:4243
10.0.1.86:4243
```
And I can connect all of them using docker info from the master:

```
root@jenkins-master:~# SLAVES="10.0.1.118:4243 10.0.1.143:4243  10.0.1.119:4243 10.0.1.86:4243"
root@jenkins-master:~# for SLAVE in $SLAVES; do docker -H tcp://$SLAVE info | grep ^Name; done
Name: jenkins-slave-5
Name: jenkins-slave-1
Name: jenkins-slave-4
Name: jenkins-slave-2
```

I tried stopping all the nodes, creating a new cluster token, starting all the slaves with the new token, and starting the manager with the new token, and only the nodes 1 and 2 are joining. Any ideas?

EDIT: Manager logs: (there is no mention of jenkins-4 and 5 on the manager, strange)

```
time="2016-01-17T21:13:22Z" level=debug msg="Updated engine state" id="N2BP:LQCL:PI3C:O6SA:4EWY:RPED:PIRH:SQNP:4VFW:ZDXQ:FVNR:445G" name=jenkins-slave-2
time="2016-01-17T21:13:37Z" level=debug msg="Updated engine state" id="BHR7:ULAU:XI4O:6Y7T:RMXE:MXM5:INTR:6OFZ:RKZP:SQR3:RZBG:SG7I" name=jenkins-slave-1
time="2016-01-17T21:14:12Z" level=debug msg="Updated engine state" id="BHR7:ULAU:XI4O:6Y7T:RMXE:MXM5:INTR:6OFZ:RKZP:SQR3:RZBG:SG7I" name=jenkins-slave-1
time="2016-01-17T21:14:19Z" level=debug msg="Updated engine state" id="N2BP:LQCL:PI3C:O6SA:4EWY:RPED:PIRH:SQNP:4VFW:ZDXQ:FVNR:445G" name=jenkins-slave-2
```

One of the slaves log:

```
time="2016-01-17T21:11:31Z" level=info msg="Registering on the discovery service every 1m0s..." addr="10.0.1.118:4243" discovery="token://xxxx"
time="2016-01-17T21:12:32Z" level=info msg="Registering on the discovery service every 1m0s..." addr="10.0.1.118:4243" discovery="token://xxxx"
time="2016-01-17T21:13:32Z" level=info msg="Registering on the discovery service every 1m0s..." addr="10.0.1.118:4243" discovery="token://xxxx"
```

**Comment by dongluochen** [comment]
@pablokbs Do you clone jenkins-slave-4/5 VMs from jenkins-slave-1/2? In that case you need to remove docker ID file `key.json` from your VM (/etc/docker/ under ubuntu) and restart docker daemon. In swarm manager, you can add debug option to get verbose logs `swarm -l "debug" manage`. Prior to swarm 1.1.0 nodes with errors are discarded after first attempt. If you restart swarm master, you should be able to catch that error. From swarm [v1.1.0-rc1](https://github.com/docker/swarm/releases/tag/v1.1.0-rc1) the nodes cannot be validated will be put at `Pending` state with explicit error. You could also try the new RC.

**Comment by pablokbs** [comment]
I didnt clone them. I'll try the debug log and let you know. Thanks!

**Comment by pablokbs** [comment]
I don't see any mention of jenkins-slave-4 nor 5 using the debug log:

```
root@jenkins-master:~# docker logs -f 93d3
time="2016-01-18T14:17:52Z" level=debug msg="Initializing discovery service" name=token uri="40cf22a03ec764686a979a7e3d078723"
time="2016-01-18T14:17:52Z" level=debug msg="Initializing strategy" name=spread
time="2016-01-18T14:17:52Z" level=debug msg="Initializing filter" name=health
time="2016-01-18T14:17:52Z" level=debug msg="Initializing filter" name=port
time="2016-01-18T14:17:52Z" level=debug msg="Initializing filter" name=dependency
time="2016-01-18T14:17:52Z" level=debug msg="Initializing filter" name=affinity
time="2016-01-18T14:17:52Z" level=debug msg="Initializing filter" name=constraint
time="2016-01-18T14:17:52Z" level=debug msg="Initializing cluster" name=swarm
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=PUT route="/containers/{name:.*}/archive"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=DELETE route="/containers/{name:.*}"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=DELETE route="/images/{name:.*}"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=DELETE route="/networks/{networkid:.*}"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=DELETE route="/volumes/{name:.*}"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=HEAD route="/containers/{name:.*}/archive"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/archive"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/top"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/exec/{execid:.*}/json"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/networks"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/images/json"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/images/viz"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/json"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/json"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/volumes/{volumename:.*}"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/get"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/ps"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/export"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/changes"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/events"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/version"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/images/search"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/images/get"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/logs"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/attach/ws"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/info"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/networks/{networkid:.*}"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/volumes"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/_ping"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/history"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/json"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/stats"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/commit"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/images/create"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/create"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/restart"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/start"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/stop"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/attach"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/auth"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/networks/create"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/volumes/create"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/exec/{execid:.*}/resize"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/rename"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/images/{name:.*}/tag"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/wait"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/resize"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/exec"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/pause"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/images/load"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/images/{name:.*}/push"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/kill"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/unpause"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/copy"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/exec/{execid:.*}/start"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/networks/{networkid:.*}/connect"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/build"
time="2016-01-18T14:17:52Z" level=debug msg="Registering HTTP route" method=POST route="/networks/{networkid:.*}/disconnect"
time="2016-01-18T14:17:52Z" level=info msg="Listening for HTTP" addr=":2375" proto=tcp
time="2016-01-18T14:17:56Z" level=debug msg="Updated engine state" id="N2BP:LQCL:PI3C:O6SA:4EWY:RPED:PIRH:SQNP:4VFW:ZDXQ:FVNR:445G" name=jenkins-slave-2
time="2016-01-18T14:17:56Z" level=info msg="Registered Engine jenkins-slave-2 at 10.0.1.86:4243"
time="2016-01-18T14:17:58Z" level=debug msg="Updated engine state" id="BHR7:ULAU:XI4O:6Y7T:RMXE:MXM5:INTR:6OFZ:RKZP:SQR3:RZBG:SG7I" name=jenkins-slave-1
time="2016-01-18T14:17:59Z" level=info msg="Registered Engine jenkins-slave-1 at 10.0.1.143:4243"
time="2016-01-18T14:18:30Z" level=debug msg="Updated engine state" id="N2BP:LQCL:PI3C:O6SA:4EWY:RPED:PIRH:SQNP:4VFW:ZDXQ:FVNR:445G" name=jenkins-slave-2
time="2016-01-18T14:18:33Z" level=debug msg="Updated engine state" id="BHR7:ULAU:XI4O:6Y7T:RMXE:MXM5:INTR:6OFZ:RKZP:SQR3:RZBG:SG7I" name=jenkins-slave-1
```

I'm using the latest image, should I try with an RC?

**Comment by dongluochen** [comment]
@pablokbs Some logs are missing. Between 
```
time="2016-01-18T14:17:52Z" level=info msg="Listening for HTTP" addr=":2375" proto=tcp
```
and 
```
time="2016-01-18T14:17:56Z" level=debug msg="Updated engine state" id="N2BP:LQCL:PI3C:O6SA:4EWY:RPED:PIRH:SQNP:4VFW:ZDXQ:FVNR:445G" name=jenkins-slave-2
```
there should be logs like: 
```
time="2016-01-19T01:39:34Z" level=info msg="Registered Engine vm3 at 192.168.xxx.xxx:4444" 
time="2016-01-19T01:39:34Z" level=info msg="Registered Engine vm7 at 192.168.xxx.xxx:4444" 
```

Can you show the commands for your swarm manager and swarm agent?

**Comment by dongluochen** [comment]
@pablokbs Do you have update on this issue? 

Swarm has released [v1.1.0-rc2](https://github.com/docker/swarm/releases/tag/v1.1.0-rc2) which contains better node management #1486. You may also try it.

**Comment by vieux** [comment]
as @dongluochen said, it should be fixed in 1.1.0.

Please reopen if you have any issue.
